text/0000-odometry-refactoring.md

```diff
@@ -0,0 +1,28 @@
+# Refactor odometry in SPG optimization problem
+
+## Summary
+[summary]: #summary
+
+Do not use `std::vector<TransformInterpolationBuffer>` in SPG optimization problem anymore.
+
+## Motivation
+[motivation]: #motivation
+
+`TransformInterpolationBuffer` does not support trimming-in-middle.
```

Comment:
I think this RFC is too fine grained and this is also not the true motivation: why do we want trimming in the middle in the first place? I agree that we should have an RFC for https://github.com/googlecartographer/cartographer/projects/1 which motivates it. @cschuet Opinion?

Reply:
You mentioned yesterday that you want to switch away from `TransformInterpolationBuffer` and follow how odometry is now kept in `MapByTime`.

Reply:
"Motivation" should be what caused us to get interested in this change. Trimming-in-the-middle is not a goal by itself, but a necessity to support life-long mapping.

Reply:
I see. If you want, I can close this PR and open an issue for handling refactoring of odometry.

Reply:
No need for an issue. We have a project already (see above) which predates the RFC process. We could add an RFC for it. Other than that, we will just do it.

Reply:
The Projects UI is pretty much read-only for non-maintainers. Where will discussion take place?

> Other than that, we will just do it.

You mean you will do the refactoring? :-)

Reply:
Yes. Already on it.